txscript/opcode.go

```diff
@@ -1850,7 +1850,17 @@ func opcodeCheckSig(op *parsedOpcode, vm *Engine) error {
 		return nil
 	}
 
-	ok := signature.Verify(hash, pubKey)
+	var ok bool
+	if vm.sigCache != nil {
+		ok = vm.sigCache.Exists(hash, signature, pubKey)
+		if !ok && signature.Verify(hash, pubKey) {
+			vm.sigCache.Add(hash, signature, pubKey)
+			ok = true
+		}
+	} else {
+		ok = signature.Verify(hash, pubKey)
+	}
```

Comment:
I disagree. As written now, both cases (`sigCache` vs no `sigCache`) are logically separated and handled distinctly.